src/main/java/core/mate/academy/service/TruckProducerImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class TruckProducerImpl implements MachineProducer<Truck> {
+    private List<Truck> trucks;
```

Comment:
Consider using access modifiers for your fields. The 'trucks' list should be private to encapsulate the internal representation and ensure that it is not modified directly from outside of the class.